lib/index.js

```diff
@@ -307,18 +352,28 @@ export function createUnifiedLanguageServer({
         /* c8 ignore next */
         if (!cwd) return
 
+        const configuration = await getDocumentSettings(textDocument.uri)
+
         const file = lspDocumentToVfile(textDocument, cwd)
 
-        const files = workspacePathToFiles.get(cwd) || []
-        workspacePathToFiles.set(cwd, [...files, file])
+        if (configuration.requireConfig) {
+          const files = workspacePathToFilesRequireConfig.get(cwd) || []
+          workspacePathToFilesRequireConfig.set(cwd, [...files, file])
+        } else {
+          const files = workspacePathToFiles.get(cwd) || []
+          workspacePathToFiles.set(cwd, [...files, file])
+        }
```

Comment:
probably better to, outside the if/else, define the map that is used with a ternary:

```
const map = configuration.requireConfig ? workspacePathToFiles : workspacePathToFilesRequireConfig
```

...and then you don’t need the if/else anymore because what’s inside can use `map`?